core/src/com/unciv/ui/tilegroups/TileSetStrings.kt

```diff
@@ -13,7 +14,10 @@ class TileSetStrings {
     val hexagon = tileSetLocation + "Hexagon"
     val crosshatchHexagon = tileSetLocation + "CrosshatchHexagon"
     val cityOverlay = tileSetLocation + "CityOverlay"
-    val railroad = tileSetLocation + "Railroad"
+    val roadsMap = RoadStatus.values()
+        .filterNot { it == RoadStatus.None }
+        .map { it to tileSetLocation + it.name }
+        .toMap()
```

Comment:
Why .toMap? Isn't this already a map?

Reply:
no, .map{} produces a List<Pair> similar to what you can feed to mapOf().

From a C++ background, you'd think "Array indexed directly by the enum", where the actual lookup could be compiled down to a single machine instruction (at least on 80xx as long as the elements are aligned at certain multiples of 4).  And indeed, `val roadsArray = RoadStatus.values().map { tileSetLocation + it.name }.toTypedArray()` and `val image = ImageGetter.getImage(tileSetStrings.roadsArray[roadStatus.ordinal])` runs fine, but - that's not a kotlin primitive Array under the hood but a java boxed Array, and the library code is full of unchecked casts, so I'm prejudiced against them...

Reply:
I've timed it - `updateRoadImages` takes an average ~770ns with the Array, ~700ns with the Map, done over 3 realistic game turns on a large map netting ~2.6M calls.